pkg/registry/core/configmap/strategy.go

```diff
@@ -86,6 +86,21 @@ func (strategy) ValidateUpdate(ctx context.Context, newObj, oldObj runtime.Objec
 	return validation.ValidateConfigMapUpdate(newCfg, oldCfg)
 }
 
+func dropDisabledFields(configMap *api.ConfigMap, oldConfigMap *api.ConfigMap) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ImmutableEphemeralVolumes) && oldConfigMap == nil {
+		configMap.Immutable = nil
```

Comment:
Will this result in silently dropping the new `Immutable` field if feature gate is disabled? If so, how about explicitly failing if field is set but feature is disabled?

Reply:
This is the pattern we use for all newly added fields, so I think it's better to be consistent.

Reply:
Ack, will leave it to API reviewers to verify

Reply:
If teh gate is disabled we want to pretend the field did not exist.  We do not error on unknown fields